### PR TITLE
Add support for sending unique reactions

### PR DESCRIPTION
**Pull request by adasiewiczr**
https://stream-io.atlassian.net/browse/CAS-471

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt
Comment (zsmb13) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:264]: `@JvmOverloads` for these?

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt
Comment (zsmb13) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt:191]: Took me a solid few minutes to figure out what kind of `event` we're talking about here - we should _really_ rename these to `Action`s at some point 😅

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/extensions/ClientExtensions.kt
Comment (zsmb13) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/extensions/ClientExtensions.kt:36]:
```suggestion
        val newScore = reactionScores.getOrElse(type) { 0 } - reactions.sumBy { it.score }
```

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelControllerImplReactionsTest.kt
Comment (zsmb13) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelControllerImplReactionsTest.kt:80]: This test name seems broken 😄

**Review by zsmb13 — COMMENTED** [review]
Looks good in general, though I don't know reactions super well, so you might want a second review here 😄 

Left notes about a few small things.

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt
Comment (adasiewiczr) [reply stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt:191]: Agree, I'll add a task for that

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/extensions/ClientExtensions.kt
Comment (adasiewiczr) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/extensions/ClientExtensions.kt:36]: nice one

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelControllerImplReactionsTest.kt
Comment (adasiewiczr) [reply stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelControllerImplReactionsTest.kt:80]: 🙈

**Review by zsmb13 — APPROVED** [review]

**Comment by zsmb13** [comment]
Approved as far as I'm concerned - again, might want a second pair of eyes on it though 😄

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt
Comment (adasiewiczr) [reply stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt:191]: https://stream-io.atlassian.net/browse/CAS-572

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelControllerImplReactionsTest.kt
Comment (ogkuzmin) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelControllerImplReactionsTest.kt:80]: My private respect for tests ❤️

**Review by ogkuzmin — APPROVED** [review]

**Pull request merged**
